### PR TITLE
Move ExponentialHistogram data to internal package.

### PR DESCRIPTION
**Pull request by anuraaga**
We almost assuredly will mark metrics SDK stable without exponential histograms in the first version. So for now, we should move the code to an internal package.

I'm hoping not, but based on https://github.com/open-telemetry/opentelemetry-specification/pull/2304#discussion_r796054799 we may do similar for Exemplar if it's not finished for metrics SDK stable release. Optimistically leaving it in place for now.

### CODE REVIEW

**Review by jamesmoessis — APPROVED** [review]
LGTM, just not sure what the [.attach_pid10327](https://github.com/open-telemetry/opentelemetry-java/pull/4217/files#diff-64825ccbf3fd1468d30d4c2a7943ea8867b9aa5b510f3b98ff30c019eb5be847) file is. Is that supposed to be there?

**Comment by jkwatson** [comment]
I approve..just need to put the "internal" scare text boilerplate on the public stuff moved in there.

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/data/exponentialhistogram/ImmutableExponentialHistogramData.java [hunk 1]
```diff
@@ -15,25 +16,32 @@
...
+  private static final ExponentialHistogramData EMPTY =
       ExponentialHistogramData.create(AggregationTemporality.CUMULATIVE, Collections.emptyList());
 
-  DoubleExponentialHistogramData() {}
+  public static ExponentialHistogramData empty() {
+    return EMPTY;
+  }
 
   /**
    * Create a DoubleExponentialHistogramData.
```
Comment (jkwatson) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/data/exponentialhistogram/ImmutableExponentialHistogramData.java:34]: javadoc drift here

**Review by jkwatson — APPROVED** [review]
looks good. a bit of javadoc drift with the renames, but other than that, good to go.

**Pull request merged**